### ISSUE TITLE
Panic when loading Nautilus Catalog generated from Databento L3 MBO Catalog

### ISSUE THREAD

**Issue by marloncalvo** [nautechsystems/nautilus_trader#3183]
# Bug Report

## Confirmation

**Before opening a bug report, please confirm:**

- [x] I’ve re-read the relevant sections of the documentation.
- [x] I’ve searched existing issues and discussions to avoid duplicates.
- [x] I’ve reviewed or skimmed the source code (or examples) to confirm the behavior is not by design.
- [x] I’ve tested this issue using a recent *development* wheel (`dev` develop or `a` nightly) and can still reproduce it.

### Expected Behavior

Generating Nautilus Catalog from Databento L3 MBO catalog, and subsequently querying Nautilus Catalog should work,

```py
from nautilus_trader.persistence.catalog import ParquetDataCatalog
from nautilus_trader.adapters.databento.loaders import DatabentoDataLoader

loader = DatabentoDataLoader()
dbn_data_loaded = loader.from_dbn_file(
    path=dbn_path,
    instrument_id=InstrumentId.from_str("PLTR.XNAS"),
    as_legacy_cython=False,
)
catalog = ParquetDataCatalog("catalog")
catalog.write_data(dbn_data_loaded)
deltas = catalog.order_book_deltas(instrument_ids=["PLTR.XNAS"])
```

### Actual Behavior

Instead, the following panic is raised from ` catalog.order_book_deltas(...)`:
```
07:23:43.467 [error] Disposing session as kernel process died ExitCode: 3221226505, Reason: 
thread 'tokio-runtime-worker' (26184) panicked at crates\\model\\src\\types\\price.rs:193:14:
Condition failed: `precision` must be 0 when `raw` is PRICE_UNDEF
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This is because we use `RecordBatch.price_precision` for `PRICE_UNDEF` cases as well - which of course will lead to mismatch between `R` (reset) and other records. For example, L3 MBO typical contains `precision = 2` but we expect `precision = 0` for `PRICE_UNDEF`. If row_0 == `R`, and row_1 == `A`, it will lead to panic.

https://github.com/nautechsystems/nautilus_trader/blob/ea4754530bcc691b639042ef553bec4c54ce98af/crates/model/src/types/price.rs#L187-L194

https://github.com/nautechsystems/nautilus_trader/blob/ea4754530bcc691b639042ef553bec4c54ce98af/crates/serialization/src/arrow/delta.rs#L223

https://github.com/nautechsystems/nautilus_trader/blob/ea4754530bcc691b639042ef553bec4c54ce98af/crates/serialization/src/arrow/delta.rs#L170-L175

@cjdsellers, I guess one fix is to assign `price = 0` for `PRICE_UNDEF` cases but don't have proper design context to validate if this is right approach.

### Repro

Here's a few Rust tests to repro this issue:
```rust
#[rstest]
fn test_query_pltr_from_existing_catalog() {
    use std::path::Path;

    let catalog_path_str = Path::new("nautilus_catalog/");

    let mut catalog = ParquetDataCatalog::new(
        catalog_path_str.to_path_buf(),
        None,
        None,
        None,
        None,
    );

    let query_result = catalog
        .query::<OrderBookDelta>(
            Some(vec!["PLTR.XNAS".to_string()]),
            None,
            None,
            None,
            None,
        )
        .unwrap();

    let deltas: Vec<Data> = query_result.collect();

    println!("Retrieved {} order book deltas for PLTR.XNAS", deltas.len());
    assert!(!deltas.is_empty(), "Should have loaded order book deltas");
}
```

```rust
  #[test]
    fn test_create_catalog_from_databento_mbo() {
        use nautilus_persistence::backend::catalog::ParquetDataCatalog;

        // User-supplied path to databento MBO data file
        let databento_path = Path::new("dbn_path");

        // Check if file exists before running test
        if !databento_path.exists() {
            println!("Skipping test: databento file not found at {:?}", databento_path);
            return;
        }

        // Arrange
        let publishers_filepath = Path::new(env!("CARGO_MANIFEST_DIR")).join("publishers.json");
        let loader = DatabentoDataLoader::new(Some(publishers_filepath)).unwrap();

        // Persistent catalog path
        let catalog_path = "nautilus_catalog/";
        println!("Creating catalog at: {:?}", catalog_path);

        // Create catalog
        let catalog = ParquetDataCatalog::from_uri(
            catalog_path,
            None,
            Some(5000),
            None,
            None,
        );

        const N: usize = 5; // Assume 'R' record is first row in DBN.
        println!("Loading top {} MBO records...", N);
        let mut deltas = Vec::new();

        for result in loader
            .read_records::<dbn::MboMsg>(
                databento_path,
                None,
                None, 
                false,
                None,
            )
            .unwrap()
            .take(N)
        {
            match result {
                Ok((Some(Data::Delta(delta)), _)) => deltas.push(delta),
                Ok(_) => continue,
                Err(e) => panic!("Error loading delta: {}", e),
            }
        }

        assert!(!deltas.is_empty(), "Should load at least one delta");
        catalog.expect("REASON").write_to_parquet(deltas, None, None, None).unwrap();
    }
```

### Specifications

- OS platform: windows/linux
- Python version: all?
- `nautilus_trader` version: lastest develop commit

**Comment by marloncalvo** [comment]
The following snippet hackily normalizes `price = 0` for `PRICE_UNDEF` cases and fixes panic:
```py
PRICE_UNDEF = 9223372036854775807

loader = DatabentoDataLoader()
dbn_data_loaded = loader.from_dbn_file(
    path=dbn_path,
    instrument_id=InstrumentId.from_str("PLTR.XNAS"),
    as_legacy_cython=True, # Must be true :)
)
for (i, record) in enumerate(dbn_data_loaded):
    if record.order.price.raw == PRICE_UNDEF:
        order = BookOrder(
            side=record.order.side,
            price=Price(value=0, precision=2),
            size=record.order.size,
            order_id=record.order.order_id,
        )
        dbn_data_loaded[i] = OrderBookDelta(
            instrument_id=record.instrument_id,
            action=record.action,
            order=order,
            flags=record.flags,
            sequence=record.sequence,
            ts_event=record.ts_event,
            ts_init=record.ts_init,
        )

catalog = ParquetDataCatalog("catalog")
catalog.write_data(dbn_data_loaded)
deltas = catalog.order_book_deltas(instrument_ids=["PLTR.XNAS"])
```

Snippet should not be used as-is as there could be some other scenarios where applying `price = 0` when `PRICE_UNDEF` is not correct

**Comment by wirrell** [comment]
Can confirm I am also experiencing this bug with Databento MBO data.

I want to add that it looks like the equivalent of `UNDEF_PRICE` from `databento_dbn` is being decoded to `170141183460469231731687303715884105727` rather than `9223372036854775807` for `R` clear orders when using Databento futures data.

e.g.

```python
import pathlib

import databento
from nautilus_trader.adapters.databento.loaders import DatabentoDataLoader

dbn_file = pathlib.Path('./databento/ICEEUCocoa/IFEU-20250221-Q8LRF8RTX8/ifeu-impact-20240821.mbo.dbn.zst'):

nautilus_db_loader = DatabentoDataLoader()

db_store = databento.DBNStore.from_file(dbn_file)
for record in store:
    if record.action == 'R':
        print(record.price)  # returns 9223372036854775807 which is databento_dbn.UNDEF_PRICE

mbo = loader.from_dbn_file(path=dbn_file, as_legacy_cython=False)
for record in mbo:
    if record.action == 'CLEAR':
            print(record.order.raw)  # returns 170141183460469231731687303715884105727
```

Not sure if this holds for equities data as I don't have access to it but wanted to highlight this discrepancy for any potential fix.

**Comment by wirrell** [comment]
Also the panic from `catalog.order_book_deltas(...)` seems to fail silently and return an empty list when calling `catalog.instruments()` if all the instruments saved in parquet files in the catalog directory have the above error.

**Comment by marloncalvo** [comment]
@wirrell, with regards to the `PRICE_UNDEF` parsed value, there's additional discrepancies I noted:
- On 128-bit high-precision compiled `nautilus_trader` package, `PRICE_UNDEF` is parsed as `170141183460469231731687303715884105727`.
- Otherwise, it's parsed as `9223372036854775807`.

Basically, `PRICE_UNDEF` is set as max value, which differs in high-precision compiled mode (default on Linux/macOS).

In my case, I had some incompatibilities transferring my catalog from Windows to Linux caused by bit-precision differences in packages from these envs. Not sure if this is at all related to your findings.

**Comment by cjdsellers** [comment]
Hi @marloncalvo and @wirrell 

Thanks for the detailed report and follow-ups.

Now fixed by checking for `PRICE_UNDEF` and `QUANTITY_UNDEF` during deserialization and applying precision 0 to those values instead of the batch precision. This avoids failing the invariant that` PRICE_UNDEF/QUANTITY_UNDEF` must have precision 0.

Also added a test for decoding deltas with UNDEF values (Clear/Reset records). Let me know if any further issues!
